packages/common/src/GenericContract.ts

```diff
@@ -0,0 +1,616 @@
+import { Buffer } from 'buffer/';
+import { stringify } from 'json-bigint';
+
+import { checkParameterLength, getContractName } from './contractHelpers';
+import { ConcordiumGRPCClient } from './GRPCClient';
+import { AccountSigner, signTransaction } from './signHelpers';
+import {
+    AccountTransactionType,
+    Base58String,
+    Base64String,
+    ContractAddress,
+    HexString,
+    InstanceInfo,
+    InvokeContractResult,
+    MakeOptional,
+    SmartContractTypeValues,
+    UpdateContractPayload,
+} from './types';
+import { AccountAddress } from './types/accountAddress';
+import { CcdAmount } from './types/ccdAmount';
+import { TransactionExpiry } from './types/transactionExpiry';
+
+/**
+ * Metadata necessary for smart contract transactions
+ */
+export type ContractTransactionMetadata = {
+    /** Amount (in microCCD) to include in the transaction. Defaults to 0n */
+    amount?: bigint;
```

Comment:
Doing this will cause breaking changes to the CIS2Contract implementation. It's a good idea along the other type wrapper suggestions, though. Will add an issue accumulating all of these suggestions, which can be scheduled with other breaking changes to be released in the near future.

Reply:
https://github.com/Concordium/concordium-node-sdk-js/issues/253 for reference.